<!DOCTYPE html>
<html lang="en">
<title>My Webpage</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<style>
body {
	background-image: url("image/images0.jpg");/* The image/images used */
		  background-color: white; /* Used if the image is unavailable */
	  height: 500px; /* You must set a specified height */
	  background-position: center; /* Center the image */
	  background-repeat: no-repeat; /* Do not repeat the image */
	  background-size: cover; /* auto |  cover | 300px 500px  -> Resize the background image to cover the entire container */

	color: maroon;
font-family: "Lato", sans-serif}
.mySlides {display: none}
</style>
<body>

<!-- Navbar -->
<div class="w3-top">
  <div class="w3-bar w3-black w3-card">
    <a class="w3-bar-item w3-button w3-padding-large w3-hide-medium w3-hide-large w3-right" href="javascript:void(0)" 
						onclick="myFunction()" title="Toggle Navigation Menu"><i class="fa fa-bars"></i></a>
    <a href="index.html" class="w3-bar-item w3-button w3-padding-large">HOME</a>  
    <div class="w3-dropdown-hover w3-hide-small">
      <button class="w3-padding-large w3-button w3-black" title="ABOUT ME">ABOUT ME <i class="fa fa-caret-down"></i></button>     
      <div class="w3-dropdown-content w3-bar-block w3-card-4">
        <a href="basic.html" class="w3-bar-item w3-button">Personal Info</a>
        <a href="myhobbies.html" class="w3-bar-item w3-button">My Hobbies</a>
        <a href="myfavfoods.html" class="w3-bar-item w3-button">My Fav Food</a>
        <a href="mygallery.html" class="w3-bar-item w3-button">MY GALLERY</a>
      </div>
    </div>
    <div class="w3-dropdown-hover w3-hide-small">
      <button class="w3-padding-large w3-button w3-black" title="PORT FOLIO">PORT FOLIO <i class="fa fa-caret-down"></i></button>     
      <div class="w3-dropdown-content w3-bar-block w3-card-4">
        <a href="skills.html" class="w3-bar-item w3-button">Professional Skills</a>
        <a href="study.html" class="w3-bar-item w3-button">Educational Info</a>
        <a href="works.html" class="w3-bar-item w3-button">Work Experiences</a>
        <a href="strengths.html" class="w3-bar-item w3-button">Strengths</a>
        <a href="activities.html" class="w3-bar-item w3-button">Activities</a>
      </div>
    </div>
    <a href="cv/cv.html" class="w3-bar-item w3-button w3-padding-large w3-hide-small">RESUME</a>
    <a href="myprojects.html" class="w3-bar-item w3-button w3-padding-large w3-hide-small">MY PROJECTS</a>
    <a href="achievements.html" class="w3-bar-item w3-button w3-padding-large w3-hide-small">ACHIEVEMENTS</a>
    <a href="index.html#contact" class="w3-bar-item w3-button w3-padding-large w3-hide-small">CONTACT</a>
    <a href="javascript:void(0)" class="w3-padding-large w3-hover-red w3-hide-small w3-right"><i class="fa fa-search"></i></a>
  </div>
</div>

<!-- Navbar on small screens (remove the onclick attribute if you want the navbar to always show on top of the content when clicking on the links) -->
<div id="navDemo" class="w3-bar-block w3-black w3-hide w3-hide-large w3-hide-medium w3-top" style="margin-top:46px">


    <a href="cv/cv.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">RESUME</a>
    <a href="myprojects.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">MY PROJECTS</a>
    <a href="achievements.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">ACHIEVEMENTS</a>
        <a href="works.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">Work Experiences</a>
        <a href="study.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">Educational Info</a>
        <a href="skills.html" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">Professional Skills</a>

  <a href="index.html#contact" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">CONTACT</a>

  <a href="#" class="w3-bar-item w3-button w3-padding-large" onclick="myFunction()">MORE</a>
</div>







<!-- Page content -->






<div class="w3-content" style="max-width:2000px;margin-top:46px">

  <div class="w3-container w3-content w3-center w3-padding-64" style="max-width:800px">
    <h2 class="w3-wide">Structural Classification of Proteins</h2> <br />  <br />  <br /> 
<br /> 
<br /> 

    <p class="w3-justify">

		Language of implementation:  Python  <br /> 
		<details>
		  <summary>Project Submission Details </summary>
			Course Title	: Thesis/Project <br /> 
			Instructor	: Amit Kumar Das, Senior Lecturer, CSE Dept <br /> 
		</details>

	
	</p>


    <p class="w3-justify">

	<br /> <b>Abstract: </b>	<br /> 
	<br /><br /><br />

The main purpose of this project is to help classify proteins based on structural sequence using Naive Bayes, Random Forest and SVM approach 
and help determine accuracy, precision, recall etc and differentiate between the classifiers.

	<br /><br /><br />
	<br /> <b>Introduction: </b>	<br /> 
	<br /><br /><br />

A protein is a polymeric macromolecule made of amino acid building blocks arranged in a linear chain and joined together by peptide bonds. 
Proteins represent the most important class of biomolecules in living organisms. 
They carry out majority of the cellular processes and act as structural constituents, catalysis agents, signaling molecules and molecular machines of every biological system. 
In all cell functions proteins are virtually involved. 
Every single protein has specific function within the body. Some of the few proteins are involved in bodily movement, while others are involved in structural support. 
Proteins vary in functions and structures. Although understanding protein structure is highly important in various areas of research (e.g. medicine and biotechnology).

<br /> <br /> 

Machine learning is a subfield of computer science that includes the study of systems that can learn from data, rather than follow only explicitly programmed instructions. 
Some of the most common techniques used for machine learning are Support Vector Machine, Artificial Neural Networks, K Nearest Neighbor and Deep learning. 
Machine learning techniques are widely used techniques in bioinformatics to solve different type of problems. 
One of the important goals followed by bioinformatics and theoretical chemistry is protein structure prediction. 
It is highly important in biotechnology and medicine. 
Different machine learning algorithms and methods have been utilized in various domains like genomics, proteomics and systems biology. 
The goal of protein structure prediction is to determine the  structure of a protein from its amino acid sequence.

<br /> <br /> 


In this paper we have tried to analyze the structure of protein and sequence generation by using different machine learning approach. 
Based on different algorithm we can find the better accuracy of different protein sequence.
In this study , different machine learning approaches have been applied to protein structure and explore the sequence of protein.




	<br /><br /><br />
	
  	<br /> <b>Related Works: </b>	<br /> 
	
	
	
	
Recently some method and system are proposed to analysis the protein and sequence of protein using machine learning method. 
In this section, we have talked about those existing methods.
Since protein is very vital element for our body structure. 
So we have discussed about the related works to find the limitation of those technologies and introduced some advanced features which will help us a lot.

	<br /><br /><br />
	
	
1. sugar-binding residue prediction system from protein sequences using support vector machine:<br />
<br />
In biological processes  the interactions between proteins and sugar chains play essential roles such as intercellular communication, immunity, and cellular recognition. 
Using machine learning algorithms numerous methods have been proposed for protein–sugar binding site prediction. 
In that paper they were classifying  sugars into acidic and nonacidic sugars and showed that their binding sites have different amino acid occurrence frequencies. 
From this result, they developed sugar-binding residue predictors committed to the two classes of sugars: an acid sugar binding predictor and a nonacidic sugar binding predictor. 
For predicting of Amino acid sequences they used only Support vector machine as a machine learning algorithm and evaluated the performance of the predictors 
using five-fold cross-validation. Though, they are not effective to learn various properties of binding site residues caused by various interactions between proteins and sugars. 
For acidic and nonacidic sugar-binding, showed the best performance in the prediction of sugar-binding residues and nonacidic sugar binding residues. 
This approach is particularly effective when the difference in the sequence features between the groups is large. 
To get better performance they can use other different machine learning algorithms. 
In few years, some of the researchers use methods utilizing glycan arrays have been developed as high throughput solutions, enabling researchers 
to obtain data on in vitro interactions between multiple sugar chains and proteins. 
Docking simulation is also a prediction method for sugar-binding residues based on their tertiary structures. 
To implement this method, many protein–ligand docking programs and molecular simulations are often employed. [9].
<br /><br />

2. Protein Structure and Sequence Similarity on the Accuracy of Machine-Learning Scoring Functions for Binding Affinity Prediction:<br />
<br />
This study is focused on structure-based protein–ligand binding affinity prediction. 
Scoring Functions (SFs) are used to predict the strength of the non-covalent interaction as binding affinity between two molecules(protein-protein or protein-DNA) 
after they have been docked. 

In this study they categorized the SFs into two broad classes: classical SFs and machine-learning SFs. 
This study shows that RF-Score is substantially more predictive than traditional X-Score, when using training sets generated with sequence similarity 
instead of structural similarity. 
That means that RF-Score is substantially more predictive than X-Score.By increasing the training set size, the number of highly similar Proteins will also increase.
So RF will produce better result. Moreover, RF-Score-v3 outperformed X-Score even with an average sized dataset.[11][12].
<br /><br />

3. Protein Residue Contacts and Prediction Methods:<br />
<br />
Many machine learning algorithms have been applied to predict protein residue contacts. 
Initial approaches to ab initio contact prediction used artificial neural networks genetic algorithm ,random forest , hidden Markov model , and support vector machines. 
Now by using of deep learning architectures with and without including correlated mutation information . Deep learning methods gives the best results.  
These machine learning-based methods use a wide range of features as input including features related to local window of the residues, information about the residue type, 
and the protein itself. 
This includes features like secondary structure, sequence profiles, solvent accessibility, mutual information of sequence profiles, residue type information, 
sequence separation length between the residues under consideration, and pairwise information between all the residues involved.[18]
<br /><br />


 

4. Prediction and characterization of human ageing-related proteins by using machine learning:<br />
There is a huge impact on human health for ageing. More than three hundred genes 
have been related to human ageing. using  state-of-the-art machine
learning methods, they classified human proteins as ageing-related or non-ageing-related. 
They  applied supervised machine learning for analysis the  ageing-related and non-ageing-related proteins .
 Support-vector machine (SVM), k-nearest neighbor (KNN), and decision tree classifiers were used for predicting ageing-related genes of the nematode on the GenAge database . 
 To classify human DNA repair genes as ageing-related or non-ageing-related they used naïve Bayes classifier and J48 decision tree. 
 Also applied three state-of-the-art machine learning tools, XGBoost ,logistic regression and support-vector machine, to classify human proteins as ageing-related 
 or non-ageing-related
The final prediction was to measure the importance of a given protein in of the human ageing process and to identify new ageing-related protein candidates.[19]
<br /><br />
5. Impact of genetic variation on three dimensional structure and function of proteins:<br />
The  PDB(Protein Data Bank)  contains atomic level three-dimensional (3D) structures of biological macromolecules (proteins, DNA, RNA). 
Knowledge of the 3D structure of the gene
product can help to understand  its function and role in disease. 
Here they observe the structural and functional changes caused by single amino acid differences, including changes in enzyme activity, 
aggregation propensity, structural stability, binding, and dissociation, context of large assemblies. 
Knowledge of the 3D structure of a gene product is helpful in predicting and understanding both function and role in disease. 
The goal of this study is to improve the understanding of the relationship between point mutations and experimentally observed consequences in 3D.
Single Nucleotide Variations (SNVs) represent the most common genetic variations observed in humans. 
In this study, they analysis small dataset to understand the consequences of a particular genetic variation at the level of the fixed protein.[15]


	<br /><br />
	
	
	
	
	
	
	
	

  	<br /> <b>Experiment: </b>	<br /> 

	<br /><br /><br />

	<br /> <i>System Requirement: </i>	<br /> 

	RAM: 4GB or higher	<br /> 
	OS: Microsoft windows 7, 8, 8.1, 10, Mac-OS	<br /> 
	Jupyter Notebook or spyder	<br /> 
	</p>
	
	<br /> <br /> 
	
    <div class="w3-row w3-padding-32">
      <div class="w3-content">
        <img src="t/t1.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t2.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
    </div>
	
	

    <p class="w3-justify">
	
	<br /> <b>Result: </b>	<br /> 

	</p>
	
	<br /><br /><br /> <h4><u>Screenshots of Result</u></h4>	<br /> <br /> <br /> 
	
    <div class="w3-row w3-padding-32">
      <div class="w3-content">
        <img src="t/t3.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t4.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t5.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t6.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t7.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t8.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t9.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
      <div class="w3-content">
        <img src="t/t10.jpg"  alt="Random Name" style="width:100%">
      </div>	<br /> 
    </div>
	
	
    <p class="w3-justify">

	From this experiment, we get highest accuracy using Random Forest approach.
SVM  gave the lowest accuracy.
Random Forest gave the most balanced result.


	<br /><br /><br />

	<br /> <b>Conclusion: </b>	<br /> 
Analysis of the structure of a protein is very important. 
A significant fact is that all tasks  can perform are based on a common principle, the twenty amino acids that can form a protein. 
That is the reason why studying proteins, their composition, structure, dynamics and function, is so important. 



	<br /> 
	<br /><br /><br />

	<br /> <b>References: </b>	<br /> <br /> <br /> 
[1] ]2017. [Online]. Available: https://en.wikipedia.org/wiki/Protein. [Accessed: 12- Apr- 2017]. <br /> 
[2] Protein structure and Function Prediction using Machine Learning Methods. 2014.<br /> 
[3] Machine Learning Approaches for the Prediction of Protein Sequences. 2018.<br /> 
[4] https://en.wikipedia.org/wiki/Protein<br /> 
 [5] ] 2018. [Online]. Available: https://en.wikipedia.org/wiki/Exploratory_data_analysis. [Accessed: 11- Apr- 2018].<br /> 
[6] Feature selection on a dataset of protein families: from exploratory data analysis to statistical variable importance", 2018<br /> 
[7] https://onlinelibrary.wiley.com/doi/full/10.1002/prot.25499/” Structural effects of point mutations in proteins,10 April,2018 <br /> 
[8] https://onlinelibrary.wiley.com/doi/abs/10.1002/prot.25505<br /> 
 [9] Masaki Banno, YusukeKomiyam, WeiCaoa, YuyaOkua, KokoroUekia, Kazuya Sumikoshia, ShugoNakamuraa, TohruTeradaa, KentaroShimizua <br /> 
“Development of a sugar-binding residue prediction system from protein sequences using support vector machine.” 23 October 2016.<br /> 
 [10] Improving intermolecular contact prediction through protein-protein interaction prediction using coevolutionary analysis with expectation-maximization. 2018.<br /> 
[11] "The Impact of Protein Structure and Sequence Similarity on the Accuracy of Machine-Learning Scoring Functions for Binding Affinity Prediction", 2018.<br /> 
[12] y. Li, "Structural and Sequence Similarity Makes a Significant Impact on Machine-Learning-Based Scoring Functions for Protein−Ligand Interactions", 2018.<br /> 
[13] Predicting functionally important residues from sequence conservation. John A. Capra and Mona Singh, 2018.<br /> 
[14] A statistical model for improved membrane protein expression using sequence-derived features. Shyam M. Saladi, Nauman Javed, Axel Müller, and William M. Clemons, Jr., 2018.
 <br /> 
[15] Impact of genetic variation on three dimensional structure and function of proteins", 2017.
 <br /> 
[16] Systematic Identification of Machine-Learning Models Aimed to Classify Critical Residues for Protein Function from Protein Structure", 2017.
 <br /> 
[17] Structural Class Classification of 3D Protein Structure Based on Multi-view 2D Images", 2018.
 <br /> 
[18] Protein Residue Contacts and Prediction Methods", 2016.
 
 <br /> 
[19] "Prediction and characterization of human ageing-related proteins by using machine learning", Csaba Kerepes, 2018.
  <br /> 

[20] https://en.wikipedia.org/wiki/Exploratory_data_analysis
 <br /> 
[21]  Improving intermolecular contact prediction through protein-protein interaction prediction using coevolutionary analysis with expectation-maximization. (2018). 
[ebook] Miguel Correa Marrero1, Richard G.H Immink2,3, Dick de Ridder1, and Aalt D.J van Dijk1,3,4. Available at: http://dx.doi.org/10.1101/254789. [Accessed 8 Jan. 2018]
 <br /> 
[22]  Ritchie DW (February 2008). "Recent progress and future directions in protein-protein docking". Current Protein & Peptide Science. 
9 (1): 1–15. doi:10.2174/138920308783565741. PMID 18336319.
 <br /> 
 [23] https://proteinstructures.com/, date 24.10.2017
 <br /> 
	
[24]. https://onlinelibrary.wiley.com/journal/10970134.”



	<br /> 

	</p>
<br /> 
<br /> 

	
	
  </div>


  
  

<br /> 
<br /> 



</div>

  
<!-- End Page Content -->









<!-- Footer -->
<footer class="w3-container w3-padding-44 w3-center w3-opacity w3-black w3-xlarge w3-bottom">
  <p class="w3-medium">My Webpage</p>
</footer>





<script>
// Automatic Slideshow - change image every 4 seconds
var myIndex = 0;
carousel();

function carousel() {
    var i;
    var x = document.getElementsByClassName("mySlides");
    for (i = 0; i < x.length; i++) {
       x[i].style.display = "none";  
    }
    myIndex++;
    if (myIndex > x.length) {myIndex = 1}    
    x[myIndex-1].style.display = "block";  
    setTimeout(carousel, 4000);    
}

// Used to toggle the menu on small screens when clicking on the menu button
function myFunction() {
    var x = document.getElementById("navDemo");
    if (x.className.indexOf("w3-show") == -1) {
        x.className += " w3-show";
    } else { 
        x.className = x.className.replace(" w3-show", "");
    }
}

// When the user clicks anywhere outside of the modal, close it
var modal = document.getElementById('ticketModal');
window.onclick = function(event) {
  if (event.target == modal) {
    modal.style.display = "none";
  }
}


</script>

</body>
</html>


